### PR TITLE
Fixing Windows CI

### PR DESCRIPTION
**Pull request by ktbarrett**

### CODE REVIEW

**Comment by themperek** [comment]
If it runs correctly, I am for it. Much nicer. 👍

**Comment by ktbarrett** [comment]
It works on my fork, but Actions are not enabled on this repo.

File: .github/workflows/functional-tests.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Functional Tests
```
Comment (imphil) [inline .github/workflows/functional-tests.yml:1]: Please add a license header.

File: .github/workflows/functional-tests.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      uses: conda-incubator/setup-miniconda@v1
+        with:
+          auto-update-conda: true
+          python-version: ${{matrix.python-version}}
+    - name: Set up cocotb build dependencies
+      run: |
+        conda install -c msys2 m2-base m2-make m2-autoconf m2-flex m2-bison m2-gperf m2w64-toolchain libpython
+    - name: Build and install Icarus
+      run: |
+        git clone https://github.com/steveicarus/iverilog.git -b ${{matrix.iverilog-version}}
```
Comment (imphil) [inline .github/workflows/functional-tests.yml:38]:
```suggestion
        git clone https://github.com/steveicarus/iverilog.git -b "${{matrix.iverilog-version}}"
```

File: .github/workflows/functional-tests.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        cd iverilog
+        bash ./autoconf.sh
+        bash ./configure
+        make -j $(nproc)
+        make install
+        cd ..
```
Comment (imphil) [inline .github/workflows/functional-tests.yml:44]: Shouldn't be necessary

File: .github/workflows/functional-tests.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          python-version: ${{matrix.python-version}}
+    - name: Set up cocotb build dependencies
+      run: |
+        conda install -c msys2 m2-base m2-make m2-autoconf m2-flex m2-bison m2-gperf m2w64-toolchain libpython
+    - name: Build and install Icarus
+      run: |
+        git clone https://github.com/steveicarus/iverilog.git -b ${{matrix.iverilog-version}}
+        cd iverilog
+        bash ./autoconf.sh
+        bash ./configure
```
Comment (imphil) [inline .github/workflows/functional-tests.yml:41]: How come those two lines need `bash`?

**Review by imphil — APPROVED** [review]
Looks good, thanks! It might be that these tests are only active after they are merged. I'm fine just merging it and fixing it up then if necessary. Or you can try pushing to a branch in the cocotb/cocotb repo, maybe that'll trigger actions from the start. (Or they didn't get triggered because the PR was in draft, I remember the logic around that changing reasonably recently.)

File: .github/workflows/functional-tests.yml (hunk 4 above)
Comment (ktbarrett) [reply .github/workflows/functional-tests.yml:41]: The first one does since it's not marked as executable. I can remove it from the second one.

**Comment by ktbarrett** [comment]
Well, it's working on [my fork](https://github.com/ktbarrett/cocotb/pull/5)...

**Comment by imphil** [comment]
Let's get it merged then and see what happens.

**Comment by imphil** [comment]
Looks good! https://github.com/cocotb/cocotb/actions

**Pull request merged**